src/PIL/ContainerIO.py

```diff
@@ -92,13 +92,13 @@ def readline(self):
 
         :returns: An 8-bit string.
         """
-        s = ""
+        s = b"" if "b" in self.fh.mode else ""
         while True:
             c = self.read(1)
             if not c:
                 break
             s = s + c
-            if c == "\n":
+            if c == (b"\n" if "b" in self.fh.mode else "\n"):
```

Comment:
This could be pulled out of the loop so it doesn't need to be recalculated every iteration of the loop.

Reply:
Good idea. I've pushed a commit for this.